src/main/java/uk/co/ramp/EvaluateCase.java

```diff
@@ -0,0 +1,230 @@
+                VirusStatus status = determineInfection(p);
+                updateVirusStatus(status, time);
+                if (status == INFECTED_SYMP) updateAlertStatus(REQUESTED_TEST, time);
+                break;
+            case INFECTED:
+                updateVirusStatus(RECOVERED, time);
+                break;
+            case INFECTED_SYMP:
+                status = determineOutcome(p);
+                updateVirusStatus(status, time);
+                break;
+            case SUSCEPTIBLE:
+            case RECOVERED:
+            case DEAD:
+                break;
+        }
+    }
+
+
+    private Set<Integer> checkUpdateAlertStatus(int time) {
+        LOGGER.trace("Changing alert status for id: {}", p.id());
+
+        p.setNextAlertStatusChange(-1);
+        switch (p.alertStatus()) {
+
+            case TESTED_POSITIVE:
+                // TODO: at present the patient doesn't move from here.. should they go to immune after some time?
+                LOGGER.trace("user {} has tested positive", p.id());
+                return alertAllContacts();
+            case AWAITING_RESULT:
+                // TODO: maybe include flag for has had virus?
+                // TODO: should a recovered person test +ve?
+                if (p.wasInfectious()) {
+                    LOGGER.trace("user {} has tested positive", p.id());
+                    updateAlertStatus(TESTED_POSITIVE, time);
+                } else {
+                    LOGGER.trace("user {} has tested negative", p.id());
+                    updateAlertStatus(NONE, time);
+                }
+
+                break;
+            case REQUESTED_TEST:
+                LOGGER.trace("user {} is awaiting test result", p.id());
+                updateAlertStatus(AWAITING_RESULT, time);
+                p.setWasInfectious(p.isInfectious());
+                break;
+            case ALERTED:
+                updateAlertStatus(REQUESTED_TEST, time);
+                LOGGER.trace("user {} has requested a test. Current Status: {}-{}", p.id(), p.status(), p.alertStatus());
+                break;
+            case NONE:
+                LOGGER.trace("user {} has been alerted. Current Status: {}-{}", p.id(), p.status(), p.alertStatus());
+                updateAlertStatus(ALERTED, time);
+                break;
+        }
+
+        return new HashSet<>();
+    }
+
+    private Set<Integer> alertAllContacts() {
+
+        Set<Integer> contactIds = new HashSet<>();
+        // add all ids
+        for (ContactRecord r : p.contactRecords()) {
+            contactIds.add(r.from());
+            contactIds.add(r.to());
+        }
+
+        // remove self
+        contactIds.remove(p.id());
+
+        LOGGER.trace("Alerting " + contactIds);
+        return contactIds;
+
+    }
+
+
+    private void updateAlertStatus(final AlertStatus newStatus, final int currentTime) {
+
+        int time = getDistributionValue(properties.meanTestTime(), properties.maxTestTime());
```

Comment:
@sam-brett We are using the DiseaseProperties distribution which is exponential here. Should we be using a linear distribution here or is this intentional? Thanks